lib/resources/df.rb

```diff
@@ -0,0 +1,28 @@
+# encoding: utf-8
+# author: Jonathan Morley
+
+module Inspec::Resources
+  class DfResource < Inspec.resource(1)
+    name 'df'
+    desc 'Use the df InSpec resource to test file system disk space usage'
+    example "
+      describe df('/') do
+        its('size') { should be >= 32000 }
+      end
+    "
+    attr_reader :partition
+
+    def initialize(partition)
+      @partition = partition
+      @cmd = inspec.command("df #{@partition} --output=size | sed \"/blocks/d; s/ *//g\"")
+    end
+
+    def size
+      @cmd.stdout.to_i
```

Comment:
So, rather than storing the `command` object in an instance var, I'd change this to something like this:

```ruby
@size ||= begin
            cmd = inspec.command("df #{@partition} --output=size | sed \"/blocks/d; s/ *//g\"").stdout
            raise Inspec::Exceptions::ResourceFailed, "Unable to get available space for partition #{partition}" if cmd.stdout.nil? || cmd.stdout.empty? || !cmd.exit_status.zero?
            cmd.stdout.to_i
          end
```

This will ensure the command is only run once (as we'll cache the size in an instance variable) and also does some additional error handling.

I'd also suggest just getting the raw output from the command and doing the string manipulation/parsing in Ruby instead of piping to `sed`.